firewood/src/merkle/node.rs

```diff
@@ -945,34 +977,26 @@ pub(super) mod tests {
         }))
     }
 
-    pub fn extension_from_hash(
-        root_hash: Option<TrieHash>,
-        encoded: Option<Vec<u8>>,
-        is_encoded_longer_than_hash_len: Option<bool>,
-        path: Vec<u8>,
-        child_address: DiskAddress,
-        child_encoded: Option<Vec<u8>>,
-    ) -> Node {
-        let inner = NodeType::Extension(ExtNode {
-            path: PartialPath(path),
-            child: child_address,
-            child_encoded,
-        });
-        Node::new_from_hash(root_hash, encoded, is_encoded_longer_than_hash_len, inner)
+    struct Nil;
+
+    impl From<Nil> for Option<&'static [u8; 0]> {
+        fn from(_val: Nil) -> Self {
+            None
+        }
     }
 
-    #[test_case(leaf(vec![0x01, 0x02, 0x03], vec![0x04, 0x05]); "leaf node")]
-    #[test_case(leaf_from_hash(None, Some(vec![0x01, 0x02, 0x03]), Some(false), vec![0x01, 0x02, 0x03], vec![0x04, 0x05]); "leaf node with encoded")]
-    #[test_case(leaf_from_hash(Some(TrieHash([0x01; 32])), None, Some(true), vec![0x01, 0x02, 0x03], vec![0x04, 0x05]); "leaf node with hash")]
+    #[test_case(leaf(vec![0x01, 0x02, 0x03], vec![0x04, 0x05]); "leaf_node")]
     #[test_case(extension(vec![0x01, 0x02, 0x03], DiskAddress::from(0x42), None); "extension with child address")]
     #[test_case(extension(vec![0x01, 0x02, 0x03], DiskAddress::null(), vec![0x01, 0x02, 0x03].into()) ; "extension without child address")]
-    #[test_case(extension_from_hash(None, Some(vec![0x01, 0x02, 0x03]), Some(false), vec![0x01, 0x02, 0x03], DiskAddress::from(0x42), None); "extension with encoded")]
-    #[test_case(extension_from_hash(Some(TrieHash([0x01; 32])), None, Some(true), vec![0x01, 0x02, 0x03], DiskAddress::from(0x42), None); "extension with hash")]
-    #[test_case(branch(0x0a, b"hello world".to_vec().into(), None); "branch with data")]
-    #[test_case(branch(0x0a, None, vec![0x01, 0x02, 0x03].into()); "branch without data")]
-    #[test_case(branch_with_hash(Some(TrieHash([0x01; 32])), None, Some(true), 0x0a, b"hello world".to_vec().into(), None); "branch with hash value")]
-    #[test_case(branch_with_hash(None, Some(vec![0x01, 0x02, 0x03]), Some(false), 0x0a, b"hello world".to_vec().into(), None); "branch with encoded")]
-    fn test_encoding(node: Node) {
```

Comment:
@xinifinity, I'm going to have to get you to explain what you're doing with these. I think you unnecessarily combined tests. We should test both with and without hashes if that makes sense. 

In any case, removing your test changes made rebasing easier. I figured I could add them back before this PR is merged.